src/JNI/com/amazonaws/kinesis/video/producer/jni/KinesisVideoClientWrapper.cpp

```diff
@@ -55,31 +55,32 @@ KinesisVideoClientWrapper::KinesisVideoClientWrapper(JNIEnv* env,
 KinesisVideoClientWrapper::~KinesisVideoClientWrapper()
 {
     STATUS retStatus = STATUS_SUCCESS;
+    JNIEnv *env;
+
+    if (mJvm == NULL) {
+        return;
+    }
+
+    mJvm->GetEnv((PVOID*) &env, JNI_VERSION_1_6);
```

Comment:
check the return val ?

Reply:
Can do in a follow-up since it is at the top of most of the methods here.

Tracking internally